lib/auth/init.go

```diff
@@ -1048,12 +1048,26 @@ type PresetUsers interface {
 	UpsertUser(ctx context.Context, user types.User) (types.User, error)
 }
 
+// getPresetUsers returns a list of all users roles expected to be available on
+// this cluster.
+func getPresetUsers() []types.User {
+	var presetUsers []types.User
+
+	accessBotUser := services.NewSystemAutomaticAccessBotUser()
+	if accessBotUser != nil {
+		presetUsers = append(presetUsers, accessBotUser)
+	}
+
+	// Certain `New$FooUser()` functions will return a nil role if the
+	// corresponding feature is disabled. They should be filtered out as they
+	// are not actually made available on the cluster.
+	return slices.DeleteFunc(presetUsers, func(r types.User) bool { return r == nil })
```

Comment:
I'm not sure I follow this block. Isn't the `if accessBotUser != nil` above enough to handle the corner cases?

Reply:
Yes, you're right, thanks.
I added the `DeleteFunc` to mimic the pattern of `getPresetRoles`, but it wasn't necessary the way I implemented `getPresetUsers`:

```go
// GetPresetRoles returns a list of all preset roles expected to be available on
// this cluster.
func GetPresetRoles() []types.Role {
	presets := []types.Role{
		services.NewPresetGroupAccessRole(),
		// ...
	}

	// Certain `New$FooRole()` functions will return a nil role if the
	// corresponding feature is disabled. They should be filtered out as they
	// are not actually made available on the cluster.
	return slices.DeleteFunc(presets, func(r types.Role) bool { return r == nil })
}
```

I've updated `getPresetUsers` to follow the same pattern, so `slices.DeleteFunc` is useful. What do you think?

```go
// getPresetUsers returns a list of all users roles expected to be available on
// this cluster.
func getPresetUsers() []types.User {
	presets := []types.User{
		services.NewSystemAutomaticAccessBotUser(),
	}

	// Certain `New$FooUser()` functions will return a nil role if the
	// corresponding feature is disabled. They should be filtered out as they
	// are not actually made available on the cluster.
	return slices.DeleteFunc(presets, func(r types.User) bool { return r == nil })
}
```

Reply:
Got it. Makes sense.